src/test/scala/chiselTests/DedupSpec.scala

```diff
@@ -81,4 +83,16 @@ class DedupSpec extends ChiselFlatSpec {
   it should "dedup modules that share a literal" in {
     assert(countModules(compile { new SharedConstantValDedupTop }) === 2)
   }
+
+  it should "not dedup modules that are in different dedup groups" in {
```

Comment:
is it supposed to be an error if you try to put a module into different dedup groups? should we show what happens if you:
      dedupGroup(top.inst0, "inst0")
      dedupGroup(top.inst0, "anothergroup")
      dedupGroup(top.inst1, "inst1")
      dedupGroup(top.inst1, "anothergroup")

Reply:
yes it should be an error, I added a test case